lib/resource_loader.dart

```diff
@@ -12,163 +10,20 @@
 ///
 library dartdoc.resource_loader;
 
-import 'dart:async' show Future;
-import 'dart:io' show Platform, File, Directory;
-import 'dart:typed_data';
-
-import 'package:http/http.dart' as http;
-import 'package:path/path.dart' as p;
-import 'package:pub_cache/pub_cache.dart';
-
-/// Optional package root path.
-String packageRootPath;
+import 'dart:async';
 
 /// Loads a `package:` resource as a String.
-Future<String> loadAsString(String path) async {
+Future<String> loadAsString(String path) {
   if (!path.startsWith('package:')) {
     throw new ArgumentError('path must begin with package:');
   }
-  Uint8List bytes = await _doLoad(path);
-  return new String.fromCharCodes(bytes);
+  return new Resource(path).readAsString();
 }
 
-/// Loads a `package:` resource as an [Uint8List].
-Future<Uint8List> loadAsBytes(String path) {
+/// Loads a `package:` resource as a [List<int>].
+Future<List<int>> loadAsBytes(String path) {
   if (!path.startsWith('package:')) {
     throw new ArgumentError('path must begin with package:');
   }
-  return _doLoad(path);
-}
-
-/// Determine how to do the load. HTTP? Snapshotted? From source?
-Future<Uint8List> _doLoad(final String path) {
```

Comment:
For now, yes. Just test this with `pub global activate` before you commit.

(you can globally activate without pushing to pub, fwiw)